vizro-core/src/vizro/models/_navigation/navigation.py

```diff
@@ -14,14 +16,13 @@ class Navigation(VizroBaseModel):
     """Navigation in [`Dashboard`][vizro.models.Dashboard] to structure [`Pages`][vizro.models.Page].
 
     Args:
-        pages (Optional[NavPagesType]): See [`NavPagesType`][vizro.models.types.NavPagesType].
-            Defaults to [].
+        pages (NavPagesType): See [`NavPagesType`][vizro.models.types.NavPagesType]. Defaults to `[]`.
         nav_selector (Optional[NavSelectorType]): See [`NavSelectorType`][vizro.models.types.NavSelectorType].
             Defaults to `None`.
     """
 
     pages: NavPagesType = []
-    nav_selector: NavSelectorType = None
+    nav_selector: Optional[NavSelectorType] = None
```

Comment:
@antonymilne - here I wasn't sure whether the `Optional` was removed intentionally?

Reply:
To be confirmed...

Reply:
After reading through this section of [the migration guide for pydantic V2](https://docs.pydantic.dev/2.0/migration/#required-optional-and-nullable-fields) and this [forum post](https://stackoverflow.com/questions/76466468/pydantic-model-fields-with-typing-optional-vs-typing-optional-none) it seems we should stick with `Optional[XXX] = None`.

To sum up the relevant parts:

**Runtime implications**
In practice this means that for the following model all fields behave identically in the sense that neither is required during initialization because all of them will receive None as their value, when no other value is provided, and the type of each of them is set to be Optional[str].

```

from typing import Optional, Union
from pydantic import BaseModel, Field

class Model(BaseModel):
    a: str | None
    b: Union[str, None]
    c: Optional[str]
    d: Optional[str] = None
    e: Optional[str] = Field(default=None)
    f: Optional[str] = Field(default_factory=lambda: None)
    g: str = None
```

It seems that the g case, the one we are talking about here is under the hood converted to Optional[str] = None:

```
Note that even g is implicitly handled in a way that sets its type to be Optional[str], even though we annotate it with just str. You can verify this by doing the following:

for field in Model.__fields__.values():
    print(repr(field))
```

What will change in pydantic V2 is the way how Optional[X] are handled if no defaults are provided as far as I undertsand e.g. `Optional[str]` and `Optional[str] = None` were both considered as non-required and a default was automatically set in V1. This is not the case anymore with V2. Optional[str] without a default will be considered as a required field, while `Optional[str] = None` is not required, can be None, and is None by default.

So I think to be more explicit, we should stick with `Optional[XXX] = None` unless I am missing something :)

Reply:
As discussed w/ @antonymilne : Changes will be reverted here and handled properly in separate PR

Reply:
See https://github.com/mckinsey/vizro/pull/210